fastapi/dependencies/utils.py

```diff
@@ -276,7 +277,7 @@ def get_dependant(
         param_field = get_param_field(
             param=param, default_field_info=params.Query, param_name=param_name
         )
-        if param_name in path_param_names:
+        if param_name in path_param_names and not issubclass(param.annotation, JsonWrapper):
             assert is_scalar_field(
```

Comment:
I think it would be better if we could rework this function and/or code path (and the corresponding one below) in such a way that `Json[X]` fields would be handled properly.

In particular, it would be better if we could test a different (closely-related) condition that didn't require specifically checking for `Json[X]`-based field types. I don't know if such a condition exists or could be easily made to exist though.

-----

It may also just make sense to loosen the restriction on what is allowed as a path/query/header-parameter type; I defer to @tiangolo about that.

------

Another preferable approach, in my opinion, would be to just change pydantic in such a way that `Json[X]` types pass the check you want to perform. I haven't thought enough about the `Json` type to know whether there is a reasonable change that could be made here.

Assuming there *is* a reasonable change, depending on what is required, it may be a breaking change requiring waiting until pydantic v2. But if such a change is possible (whether reasonable or not), there *may* be a simple way to accomplish your goal by subclassing the `Json` class and changing an attribute (you could call the subclass `QueryJson` or similar).

Reply:
Is @tiangolo maintainer there as well which could make an change to pydantic for fastapi realistic?
In any case I'm not sure if Pydantic is the correct place needing a change.
After all Pydantic is handling the Json type very well, decoding json, and reading it into lists, dicts and if specified even into real models just fine.
Currently we are throwing away that capability by limiting it to just the Body as source type.

It's just FastAPI adding too restrictive code on top of pydantic, therefore I made an exception for the case we want to handle here, Json.

It might be possible to maybe loosen the existing code in a more general way, however I don't really understand the existing code _(a few comments here and there would be really nice!)_ to make any in depth changes.